services/provider/server/server_test.go

```diff
@@ -0,0 +1,343 @@
+				},
+			},
+			State: ocsv1alpha1.StorageConsumerStateReady,
+		},
+	}
+
+	consumerResource1 = &ocsv1alpha1.StorageConsumer{
+		ObjectMeta: metav1.ObjectMeta{Name: "consumer1", UID: "uid1"},
+		Status: ocsv1alpha1.StorageConsumerStatus{
+			State: ocsv1alpha1.StorageConsumerStateFailed,
+		},
+	}
+
+	consumerResource2 = &ocsv1alpha1.StorageConsumer{
+		ObjectMeta: metav1.ObjectMeta{Name: "consumer2", UID: "uid2"},
+		Status: ocsv1alpha1.StorageConsumerStatus{
+			State: ocsv1alpha1.StorageConsumerStateConfiguring,
+		},
+	}
+	consumerResource3 = &ocsv1alpha1.StorageConsumer{
+		ObjectMeta: metav1.ObjectMeta{Name: "consumer3", UID: "uid3"},
+		Status: ocsv1alpha1.StorageConsumerStatus{
+			State: ocsv1alpha1.StorageConsumerStateDeleting,
+		},
+	}
+	consumerResource4 = &ocsv1alpha1.StorageConsumer{
+		ObjectMeta: metav1.ObjectMeta{Name: "consumer4", UID: "uid4"},
+		Status:     ocsv1alpha1.StorageConsumerStatus{},
+	}
+)
+
+func TestGetJSONConnectionDetails(t *testing.T) {
+	ctx := context.TODO()
+	objects := []runtime.Object{
+		&v1.ConfigMap{},
+		&v1.Secret{},
+		&v1.Service{},
+		&rookCephv1.CephClient{},
+		consumerResource,
+		consumerResource1,
+		consumerResource2,
+		consumerResource3,
+		consumerResource4,
+	}
+
+	client := newFakeClient(t, objects...)
+	consumerManager, err := newConsumerManager(ctx, client, "openshift-storage")
+	assert.NoError(t, err)
+
+	_, err = consumerManager.Create(ctx, "consumer", "ticket", resource.MustParse("1G"))
+	assert.NoError(t, err)
+
+	server := &OCSProviderServer{
+		client:    client,
+		namespace: "openshift-storage",
+		consumerManager: &ocsConsumerManager{
+			nameByUID: map[types.UID]string{
+				consumerResource.UID: consumerResource.Name,
+			},
+			client: client,
+		},
+	}
+
+	for _, i := range consumerResource.Status.CephResources {
+		if i.Kind == "CephClient" {
+			cephClient, secret := createCephClientAndSecret(i.Name, server)
+			cephClient.Status = &rookCephv1.CephClientStatus{
+				Info: map[string]string{
+					"secretName": fmt.Sprintf("rook-ceph-client-%s", i.Name),
+				},
+			}
+
+			assert.NoError(t, client.Create(ctx, cephClient))
+			assert.NoError(t, client.Create(ctx, secret))
+		}
+	}
+
+	monCm, monSc, mgrSvc := createMonConfigMapandSecret(server)
+	assert.NoError(t, client.Create(ctx, monCm))
+	assert.NoError(t, client.Create(ctx, monSc))
+	assert.NoError(t, client.Create(ctx, mgrSvc))
+
+	// When ocsv1alpha1.StorageConsumerStateReady
+	req := pb.StorageConfigRequest{
+		StorageConsumerUUID: string(consumerResource.UID),
+	}
+	storageConRes, err := server.GetStorageConfig(ctx, &req)
+	assert.NoError(t, err)
+	assert.NotNil(t, storageConRes)
+
+	for i := range mockExtR {
+		data, err := json.Marshal(mockExtR[i].Data)
+		assert.NoError(t, err)
+		assert.Equal(t, storageConRes.ExternalResource[i].Data, data)
+		assert.Equal(t, storageConRes.ExternalResource[i].Kind, mockExtR[i].Kind)
+		assert.Equal(t, storageConRes.ExternalResource[i].Name, mockExtR[i].Name)
+	}
```

Comment:
This tests only the case where there are CephObjects on the storage consumer, 
I think it will also benefit us testing the case where there aren't any